app/Picture.py

```diff
@@ -0,0 +1,20 @@
+from PIL import Image
+
+
+class Picture:
+    def __init__(self) -> None:
+        self.image = None
```

Comment:
IMO these instance attributes should all be annotated, since `self.image` is only of `NoneType` without telling the typechecker otherwise.

It can also be annotated in the form
```py
class Picture:
    if TYPE_CHECKING:
        image: Image.Image | None
        is_solved: bool
        tiles: dict[?, Tile]  # replace ? with the actual type
        tile_order = list[?]  # I do not know what type this is either

Reply:
Yes, I agree. Type annotation is important. My other PR has a lot more trouble with that. I think we need to discuss that more, because there are a few thing I don't understand, like how to annotate correctly and still get autocomplete suggestions in PyCharm, or PyCharm not complaining about something else.